src/forms/services/form-submission.service.ts

```
@@ -1030,4 +1055,320 @@ export class FormSubmissionService {
+      const userSource = userDoc._source as IUser;
+      const applications = userSource.applications || [];
+
+      // Use the actual submissionId and formId from the updatedSubmission
+      const formIdToMatch = updatedSubmission.formId;
+      const submissionIdToMatch = updatedSubmission.submissionId;
+
+      // --- NEW LOGIC: Fetch form schema and build fieldId -> pageName map ---
+      let fieldIdToPageName: Record<string, string> = {};
+      try {
+        const form = await this.formsService.getFormById(formIdToMatch);
+        const fieldsObj = form && form.fields ? (form.fields as any) : null;
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)